core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -138,9 +145,32 @@ public Environment(Settings settings) {
                 dataFiles[i] = PathUtils.get(dataPaths.get(i));
                 dataWithClusterFiles[i] = dataFiles[i].resolve(clusterName.value());
             }
+            if (DEFAULT_PATH_DATA_SETTING.exists(settings)) {
+                final String defaultPathDataValue = DEFAULT_PATH_DATA_SETTING.get(settings);
+                final Set<Path> dataFilesSet = Arrays.stream(dataFiles).collect(Collectors.toSet());
+                final Path defaultPathData = PathUtils.get(defaultPathDataValue);
+                if (dataFilesSet.size() == 1 && dataFilesSet.contains(defaultPathData)) {
+                    // default path data was used to set path data
+                    this.defaultPathData = null;
+                } else if (dataFilesSet.contains(defaultPathData)) {
+                    final String message = String.format(
+                            Locale.ROOT,
+                            "do not include default.path.data [%s] in path.data %s",
```

Comment:
The output will show that it is an array already because we're using `Arrays.toString(dataFiles)` here (so we get the brackets, etc.).